lib/Parser/ptree.h

```diff
@@ -600,9 +600,21 @@ struct PnFinally : PnStmt
     ParseNodePtr pnodeBody;
 };
 
+enum JsType : byte 
+{
+	t_object,
+	t_string,
+	t_int,
+	t_float,
+	t_bool,
+	t_array,
+	t_function
+};
+
 struct ParseNode
 {
     OpCode nop;
+    JsType typeHint; //Not sure if putting it here is the best idea
```

Comment:
i think it's ok to put in here. one nit is that we should move this def to be near bools, otherwise struct will use extra memory due to padding. usually not that important to worry about, but functions will have very many parse nodes, so a few bytes saved can add up